lib/resty/upload.lua

```diff
@@ -93,21 +111,33 @@ function _M.set_timeout(self, timeout)
     return sock:settimeout(timeout)
 end
 
-
 local function discard_line(self)
     local read_line = self.read_line
 
     local line, err = read_line(self.line_size)
     if not line then
+        if self.restore_body_buffer then
+            ngx_finish_body()
+        end
         return nil, err
     end
 
+    if self.restore_body_buffer then
+        ngx_append_body(line .. "\r\n")
```

Comment:
I think this is fine, but you could also have:
```lua
ngx_append_body(line)
ngx_append_body("\r\n")
```

Reply:
I don't see any point in not concatenating the string here, the same goes for the rest of the remarks regarding concatenation.